app/models/thredded/null_user.rb

```diff
@@ -34,6 +40,8 @@ def thredded_anonymous?
       true
     end
 
+    alias_method :anonymous?, :thredded_anonymous?
```

Comment:
Throughout the thredded engine we make use of the null object pattern and return either `super` (from the inherited ApplicationController), or OUR (thredded's) NullUser. See [here](https://github.com/thredded/thredded/blob/8dedc824487063e2cc989c29af0b231c9c9485cd/app/controllers/thredded/application_controller.rb#L91-L93). The issue with this is that the implementer *does not* own that NullUser class. Therefore the parent application controller might be sending methods to an object it does not control.

Here's an example ApplicationController snippet from my parent class.

```ruby
class ApplicationController < ActionController::Base
  def user_signed_in?
    current_user.present? && !current_user.anonymous?
    # I could really easily fix this if I change `anonymous` to `thredded_anonymous?`
  end
end
```

I suppose it's easy enough to implement `thredded_anonymous?` onto my parent User class but I think the less onus we put on the implementer, the better. Also - asking the parent class to lean heavily on a namedspaced method in a dependency feels a little weird, no?

What do you think? (By the way - I don't know if I was entirely clear--I may have rambled a bit there--so let me know if I'm missing any details.)